.github/workflows/build-and-run-batch-job.yaml

```diff
@@ -1,26 +1,58 @@
-# Workflow that builds a Docker image containing the model code,
-# pushes it to the GitHub Container Registry, and then optionally uses
-# that container image to run the model using an AWS Batch job.
+# Reusable workflow that builds a Docker image, pushes it to the GitHub
+# Container Registry, and then optionally uses that container image to run
+# an AWS Batch job.
 #
-# Images are built on every commit to a PR or main branch in order to ensure
-# that the build continues to work properly, but Batch jobs are gated behind
-# a `deploy` environment that requires manual approval from a
-# @ccao-data/core-team member.
-
-name: deploy
+# Requirements:
+#
+#   * A Dockerfile must be defined in the root of the repo whose workflow is
+#     calling this one.
+#   * A `deploy` environment must be configured in the calling repo. This
+#     environment is used to protect the `run` job, which must be approved by
+#     a core team member before it will run.
+#   * The calling workflow must grant the following permissions to the job
+#     that calls this workflow:
+#       * contents: read
+#       * id-token: write
+#       * packages: write
+#   * Various required inputs and secrets, documented below, must be passed in
+#     by the calling workflow.
+
+name: build-and-run-batch-job
 
 on:
-  pull_request:
-  workflow_dispatch:
-  push:
-    branches: [master]
+  workflow_call:
```

Comment:
By switching to the `workflow_call` event, we mark this workflow as reusable. More on reusable workflows [here](https://docs.github.com/en/actions/using-workflows/reusing-workflows).